### PR TITLE
[ncp] integrate netif isUp state update

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR integrates the `Netif::SetNetifState` with `NcpSpinel` so that when the netif state on NCP changes, the wpan state on the host updates accordingly.

### CODE REVIEW

File: src/ncp/ncp_spinel.cpp [hunk 1]
```diff
@@ -72,7 +72,9 @@ void NcpSpinel::Init(ot::Spinel::SpinelDriver &aSpinelDriver, PropsObserver &aOb
 
 void NcpSpinel::Deinit(void)
 {
-    mSpinelDriver = nullptr;
+    mSpinelDriver              = nullptr;
+    mIp6AddressTableCallback   = nullptr;
+    mNetifStateChangedCallback = nullptr;
```
Comment (zhanglongxia) [inline src/ncp/ncp_spinel.cpp:77]: Do we need to initialize these variables to `nullptr` in the constructor?

Comment (Irving-cl) [reply src/ncp/ncp_spinel.cpp:77]: The `std::function` has default constructor to let the member be `nullptr`. So I think it's not necessary to initialize them explicitly. Thoughts?

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**